lib/streamlit/commands/execution_control.py

```diff
@@ -139,6 +139,7 @@ def switch_page(page: str) -> NoReturn:  # type: ignore[misc]
         raise NoSessionContext()
 
     main_script_path = os.path.join(os.getcwd(), ctx.main_script_path)
+    main_script_path = os.path.normpath(main_script_path)
     main_script_directory = os.path.dirname(main_script_path)
```

Comment:
nit: these lines could be put into a shared utils method, e.g. `resolve_script_path()`. This would make it also possible to unit test this aspect.